import_handlers.js

```diff
@@ -0,0 +1,154 @@
+        if (!date_moment.isValid()) {
+            console.log('Not a valid date:', date);
+            continue
+        }
+
+        // Create transaction objects
+        transactions.push(new Transaction(date_moment, from, to, narrative, amount));
+    }
+
+    // Use the transactions just loaded to update the accounts
+    updateAccounts(accounts, transactions);
+    return transactions;
+}
+
+function import_json_data(inputPath, accounts) {
+    let transactions = [];
+
+    const json_str_out = fs.readFileSync(inputPath, 'utf8');
+    const json = JSON.parse(json_str_out);
+
+    json.forEach(function (t) {
+        const date_moment = moment(t['Date'], 'YYYY-MM-DD');
+        if (!date_moment.isValid()) {
+            console.log('Not a valid date', date_moment);
+            return;
+        }
+
+        transactions.push(new Transaction(date_moment,
+                                          t["FromAccount"],
+                                          t["ToAccount"],
+                                          t["Narrative"],
+                                          t["Amount"]));
+    });
+
+    updateAccounts(accounts, transactions);
+    return transactions;
+}
+
+function import_xml_data(inputPath, accounts) {
+    let transactions = [];
+
+    const xml_str_out = fs.readFileSync(inputPath, 'utf8');
+    var parseString = xml2js.parseString;
+    parseString(xml_str_out, function (err, result) {
+        // Convert XML to JSON
+        const as_json = JSON.parse(JSON.stringify(result));
+        const ts = as_json["TransactionList"]["SupportTransaction"];
+        ts.forEach(function(t) {
+            const amount = t['Value'][0];
+            
+            // Check valid amounts (with isNaN)
+            if (isNaN(parseFloat(amount))) {
+                console.log('Not a valid amount');
+                return;
+            }
+
+            const nDaysSince = parseInt(t['$']['Date']); // The attribute of a SupportTransaction;
+            const date_moment = moment('1900-01-01', 'YYYY-MM-DD').add('days', nDaysSince);
+            if (!date_moment.isValid()) {
+                console.log('Not a valid date', date_moment);
+                return;
+            }
+            transactions.push(new Transaction(date_moment,
+                                              t['Parties'][0]['From'][0],
+                                              t['Parties'][0]['To'][0],
+                                              t['Description'][0],
+                                              amount));
+        });
+    });
+    
+    updateAccounts(accounts, transactions);
+    return transactions;
+}
+
+// Once all the transactions have been read into the transactions array
+// Update the accounts array with money transfers
+function updateAccounts(accounts, transactions) {
+    transactions.forEach(function(element) {
+        // For each transaction:
+    
+        [element.from, element.to].forEach(function(name) {

```

Comment:
I'd be inclined to invert this slightly; Give your anonymous function a name (e.g. `storeTransactionInAccount` and then have the body of your `forEach` just be `storeTransactionInAccount(element, element.from); storeTransactionInAccount(element, element.to)`. I'd also go with naming it `transaction` rather than `element` :)

Up to you whether you want to define `storeTransactionInAccount` at the top scope (taking two arguments) or inside the `forEach` (taking one argument, since `element` is in scope for the closure).